nebula_hw_interfaces/include/nebula_hw_interfaces/nebula_hw_interfaces_common/connections/udp.hpp

```diff
@@ -0,0 +1,383 @@
+   *
+   * @param sender_ip The only allowed sender IP. Cannot be a multicast or broadcast address.
+   * @param sender_port The only allowed sender port.
+   */
+  UdpSocket & limit_to_sender(const std::string & sender_ip, uint16_t sender_port)
+  {
+    if (state_ >= State::ACTIVE) throw UsageError("Sender has to be set before subscribing");
+
+    sender_.emplace(Endpoint{sender_ip, sender_port});
+    return *this;
+  }
+
+  /**
+   * @brief Set the MTU this socket supports. While this can be set arbitrarily, it is best set to
+   * the MTU of the network interface, or to the maximum expected packet length.
+   *
+   * @param bytes The MTU size. The default value is 1500.
+   */
+  UdpSocket & set_mtu(size_t bytes)
+  {
+    if (state_ >= State::ACTIVE) throw UsageError("MTU size has to be set before subscribing");
+
+    buffer_size_ = bytes;
+    return *this;
+  }
+
+  UdpSocket & set_socket_buffer_size(size_t bytes)
+  {
+    if (state_ > State::INITIALIZED) throw UsageError("Buffer size has to be set before binding");
+    if (bytes > static_cast<size_t>(INT32_MAX))
+      throw UsageError("The maximum value supported (0x7FFFFFF) has been exceeded");
+
+    auto buf_size = static_cast<int>(bytes);
+    int result = setsockopt(sock_fd_, SOL_SOCKET, SO_RCVBUF, &buf_size, sizeof(buf_size));
+    if (result < 0) throw SocketError(errno);
+    return *this;
+  }
+
+  /**
+   * @brief Join an IP multicast group. Only one group can be joined by the socket.
+   *
+   * @param group_ip The multicast IP. It has to be in the multicast range `224.0.0.0/4` (between
+   * `224.0.0.0` and `239.255.255.255`).
+   */
+  UdpSocket & join_multicast_group(const std::string & group_ip)
+  {
+    if (state_ < State::INITIALIZED) throw UsageError("Socket has to be initialized first");
+    if (state_ >= State::BOUND)
+      throw UsageError("Multicast groups have to be joined before binding");
+
+    ip_mreq mreq;
+    mreq.imr_multiaddr.s_addr = inet_addr(group_ip.c_str());  // Multicast group address
+    mreq.imr_interface.s_addr = inet_addr(host_.ip.c_str());
+
+    int result = setsockopt(sock_fd_, IPPROTO_IP, IP_ADD_MEMBERSHIP, &mreq, sizeof(mreq));
+    if (result < 0) throw SocketError(errno);
+
+    multicast_ip_.emplace(group_ip);
+    return *this;
+  }
+
+  /**
+   * @brief Bind the socket to host IP and port given in `init()`. If `join_multicast_group()` was
+   * called before this function, the socket will be bound to `group_ip` instead. At least `init()`
+   * has to have been called before.
+   */
+  UdpSocket & bind()
+  {
+    if (state_ < State::INITIALIZED) throw UsageError("Socket has to be initialized first");
+    if (state_ >= State::BOUND) throw UsageError("Re-binding already bound socket");
+
+    sockaddr_in addr{};
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(host_.port);
+    addr.sin_addr.s_addr = multicast_ip_ ? inet_addr(multicast_ip_->c_str()) : INADDR_ANY;
```

Comment:
I agree, would be good to have.
Generally, in Nebula, we should probably disallow the use of the IP broadcast address (`255.255.255.255`) since that is almost never a good idea but somehow became the default in our configurations. What do you think?

Reply:
Agre 1000%. I think LiDAR manufacturers usually use it because it's easier out of the box, but the service cost is high with having it as default.

Reply:
Addressed in d60855f.